squidpy/pl/_color_utils.py

```diff
@@ -0,0 +1,53 @@
+"""Utils for plotting functions."""
+from __future__ import annotations
+
+from typing import Any, Union, Mapping, Sequence
+
+from scanpy import logging as logg
+from anndata import AnnData
+from scanpy.plotting._utils import add_colors_for_categorical_sample_annotation
+
+import numpy as np
+
+from matplotlib.colors import to_hex, ListedColormap
+import matplotlib.pyplot as plt
+
+from squidpy._constants._pkg_constants import Key
+
+Palette_t = Union[str, ListedColormap, None]
+
+
+def _maybe_set_colors(source: AnnData, target: AnnData, key: str, palette: str | None = None) -> None:
+    color_key = Key.uns.colors(key)
+    try:
+        if palette is not None:
+            raise KeyError("Unable to copy the palette when there was other explicitly specified.")
+        target.uns[color_key] = source.uns[color_key]
+    except KeyError:
+        add_colors_for_categorical_sample_annotation(target, key=key, force_update_colors=True, palette=palette)
+
+
+def _get_palette(
+    adata: AnnData, cluster_key: str, categories: Sequence[Any], palette: Palette_t = None
+) -> Mapping[str, str] | None:
+    if palette is None:
```

Comment:
Seems also natural to handle `palette` as a `Sequence[str | NDArrayA]`, but low prio

Reply:
ok will leave it then